RockLib.Logging/LogProviders/ILogLevelResolver.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RockLib.Logging.LogProviders
+{
+    /// <summary>
+    /// Used to resolve the <see cref="LogLevel"/> when logging with an <see cref="ILogger"/>
+    /// </summary>
+    public interface ILogLevelResolver
+    {
+        /// <summary>
+        /// Retrieves the currently active <see cref="LogLevel"/>
+        /// </summary>
+        /// <returns>The current <see cref="LogLevel"/> or null if it cannot be determined</returns>
+        /// <remarks>If the <see cref="LogLevel"/> cannot be determined, and is returned as null, then the default <see cref="LogLevel"/> 
+        /// of the <see cref="ILogger"/> will be used instead</remarks>
+        LogLevel? GetLogLevel();
```

Comment:
After thinking about this a bit, I decided that allowing `null` to be returned would be ideal. Say you're in a scenario where your log level is determined by a downstream dependency that is not currently available.. you need to be able to fallback on something - and that something should be able to be the default log level as determined by your configuration settings (i.e. your `appsettings.json` file)